drivers/watchdog/wdt_gecko.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2019 Interay Solutions B.V.
+ * Copyright (c) 2019 Oane Kingma
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/watchdog.h>
+#include <em_wdog.h>
+#include <em_cmu.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_gecko);
+
+/* Timeout period in cycles */
+static const u32_t timeout_in_cycles_tbl[16] = {
+	9, 17, 33, 65, 129, 257, 513, 1025, 2049, 4097,
+	8193, 16385, 32769, 65537, 131073, 262145
+};
+
+/* Device constant configuration parameters */
+struct wdt_gecko_cfg {
+	WDOG_TypeDef *base;
+	void (*irq_cfg_func)(void);
+};
+
+struct wdt_gecko_data {
+	wdt_callback_t callback;
+	WDOG_Init_TypeDef wdog_config;
+	bool timeout_valid;
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+#define DEV_DATA(dev) \
+	((struct wdt_gecko_data *)(dev)->driver_data)
+#define DEV_CFG(dev) \
+	((struct wdt_gecko_cfg *)(dev)->config->config_info)
+
+static int wdt_gecko_convert_timeout(u32_t timeout)
+{
+	int idx = 0;
+
+	/* When using ULFRCO (default), 1 cycle is 1 ms +/- 12% */
+	while (idx < ARRAY_SIZE(timeout_in_cycles_tbl)) {
+		if (timeout > timeout_in_cycles_tbl[idx]) {
+			idx++;
+			continue;
+		}
+
+		/* Found the (rounded up?) value of cycles for
+		 *  supplied timeout
+		 */
+		break;
+	}
+
+	return idx;
+}
```

Comment:
I propose to write this part as:
```
/* Defines maximum WDOG_CTRL.PERSEL value which is used by the watchdog module
 * to select its timeout period.
 */
#define WDT_GECKO_MAX_PERIOD_SELECT_VALUE 15

static u32_t wdt_gecko_get_timeout_from_persel(int perSel)
{
	return (8 << perSel) + 1;
}

/* Find the rounded up value of cycles for supplied timeout. When using ULFRCO
 * (default), 1 cycle is 1 ms +/- 12%.
 */
static int wdt_gecko_get_persel_from_timeout(u32_t timeout)
{
	int idx;

	for (idx = 0; idx < WDT_GECKO_MAX_PERIOD_SELECT_VALUE; idx++) {
		if (wdt_gecko_get_timeout_from_persel(idx) >= timeout) {
			break;
		}
	}

	return idx;
}
```
and remove `timeout_in_cycles_tbl` table. This implementation should reduce driver code size.

Reply:
I'll change it, looks cleaner.